tests/testthat/test-as_gt.R

```diff
@@ -0,0 +1,290 @@
+  # header
+  expect_equal(
+    lapply(
+      my_spanning_tbl$table_styling$header$interpret_label,
+      \(x) do.call(eval(parse(text = x)), list("")) |> class()
+    ),
+    lapply(gt_spanning_tbl$`_boxhead`$column_label, class)
+  )
+
+  # spanning header
+  expect_equal(
+    sapply(
+      my_spanning_tbl$table_styling$header |>
+        dplyr::filter(!is.na(spanning_header)) |>
+        dplyr::pull(interpret_spanning_header),
+      \(x) do.call(eval(parse(text = x)), list("")) |> class()
+    ),
+    gt_spanning_tbl$`_spanners`$spanner_label[[1]] |> class() |>
+      rep(length(gt_spanning_tbl$`_spanners`$vars[[1]])),
+    ignore_attr = "names"
+  )
+
+  # customize interpreter
+  tbl <- my_spanning_tbl |>
+    modify_table_styling(columns = "stat_1", label = "Stat I", spanning_header = "MyTest", text_interpret = "html")
+  gt_tbl <- tbl |> as_gt()
+  if ("html" %in% names(attributes(gt_tbl$`_spanners`$spanner_label[[1]]))) {
+    attr(gt_tbl$`_spanners`$spanner_label[[1]], "html") <- c("html", "character")
+  }
+
+  # header
+  expect_equal(
+    lapply(
+      tbl$table_styling$header$interpret_label,
+      \(x) do.call(eval(parse(text = x)), list("")) |> class()
+    ),
+    lapply(gt_tbl$`_boxhead`$column_label, class)
+  )
+
+  # spanning header
+  expect_equal(
+    sapply(
+      tbl$table_styling$header |>
+        dplyr::filter(!is.na(spanning_header)) |>
+        dplyr::pull(interpret_spanning_header),
+      \(x) do.call(eval(parse(text = x)), list("")) |> class()
+    ),
+    attributes(gt_tbl$`_spanners`$spanner_label[[1]]),
+    ignore_attr = "names"
+  )
+})
+
+test_that("as_gt passes table footnotes & footnote abbreviations correctly", {
+  # footnote
+  expect_equal(
+    my_tbl_summary$table_styling$footnote$column,
+    gt_tbl_summary$`_footnotes`$colname |> unique()
+  )
+  expect_equal(
+    my_tbl_summary$table_styling$footnote$footnote,
+    gt_tbl_summary$`_footnotes`$footnotes |> unlist() |> unique()
+  )
+
+  tbl_fa <- my_tbl_summary |>
+    modify_footnote(stat_0 = "N = number of observations", abbreviation = TRUE)
+  gt_tbl_fa <- tbl_fa |> as_gt()
+
+  # footnote_abbrev
+  vis_cols <- tbl_fa$table_styling$header |>
+    dplyr::filter(hide == FALSE) |>
+    dplyr::select(column) |>
+    unlist()
+  footnotes_vis <- rbind(
+    tbl_fa$table_styling$footnote,
+    tbl_fa$table_styling$footnote_abbrev
+  ) |>
+    dplyr::filter(column %in% vis_cols) |>
+    dplyr::arrange(rows)
+  expect_equal(
+    footnotes_vis$column,
+    gt_tbl_fa$`_footnotes` |>
+      dplyr::distinct(pick(!any_of("rownum"))) |>
+      dplyr::arrange(locnum) |>
+      dplyr::pull(colname)
+  )
+  expect_equal(
+    footnotes_vis$footnote |> unname(),
+    gt_tbl_fa$`_footnotes` |>
+      dplyr::distinct(pick(!any_of("rownum"))) |>
+      dplyr::arrange(locnum) |>
+      dplyr::pull(footnotes) |>
+      unlist()
+  )
+})
+
+test_that("as_gt passes table indentation correctly", {
```

Comment:
Can we add a test where indentation has been removed?

Reply:
Added!